File: 프로그래머스/lv2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.py
def solution(b):
    R = len(b)
    C = len(b[0])
    
    for i in range(1, R):
        for j in range(1, C):
            if b[i][j] == 1:
                b[i][j] = min(b[i-1][j], b[i-1][j-1], b[i][j-1]) + 1
    
    answer = 0
    for i in range(R):
        temp = max(b[i])
        answer = max(answer, temp)
    
    return answer**2                
    
            

File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.java
class Solution {
    public int solution(int[][] board) {
        int N = board.length;
        int M = board[0].length;

        int answer = 0;
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < M; j++) {
                if (i == 0 || j == 0) {
                    answer = Math.max(answer, board[i][j]);
                    continue;
                }
                
                if (board[i][j] == 0) {
                    continue;
                }
                board[i][j] = Math.min(Math.min(board[i - 1][j], board[i][j - 1]), board[i - 1][j - 1]) + 1;
                int v = board[i][j] * board[i][j];
                answer = Math.max(answer, v);
            }
        }

        return answer;
    }
}